internal/pkg/cli/dialog/template_inputs.go

```diff
@@ -0,0 +1,282 @@
+				lines.WriteString("\n")
+			}
+		}
+	}
+
+	return lines.String()
+}
+
+// detectInputs - detects potential inputs in each config and config row.
+func (d *templateInputsDialog) detectInputs() {
+	d.fieldsByObject = make(map[model.Key]inputFields)
+	for _, c := range d.configs {
+		d.detectInputsFor(c.ConfigKey, c.ComponentId, c.Content)
+		for _, r := range c.Rows {
+			d.detectInputsFor(r.ConfigRowKey, c.ComponentId, r.Content)
+		}
+	}
+}
+
+// detectInputsFor config or config row.
+func (d *templateInputsDialog) detectInputsFor(objectKey model.Key, componentId model.ComponentId, content *orderedmap.OrderedMap) {
+	content.VisitAllRecursive(func(fieldPath orderedmap.Key, value interface{}, parent interface{}) {
+		// Root key must be "parameters"
+		if len(fieldPath) < 2 || fieldPath.First() != orderedmap.MapStep("parameters") {
+			return
+		}
+
+		// Must be object field
+		fieldKey, isObjectField := fieldPath.Last().(orderedmap.MapStep)
+		if !isObjectField {
+			return
+		}
+
+		isSecret := encryptionapi.IsKeyToEncrypt(string(fieldKey))
+
+		// Detect type, kind and default value
+		var inputType input.Type
+		var inputKind input.Kind
+		var inputOptions input.Options
+		var defaultValue interface{}
+		valRef := reflect.ValueOf(value)
+		switch valRef.Kind() {
+		case reflect.String:
+			inputType = input.TypeString
+			if isSecret {
+				inputKind = input.KindHidden
+			} else {
+				inputKind = input.KindInput
+			}
+
+			// Use as default value, if it is not a secret
+			if !isSecret && len(value.(string)) > 0 {
+				defaultValue = value
+			}
+		case reflect.Int:
+			inputType = input.TypeInt
+			inputKind = input.KindInput
+			if !isSecret && value.(int) != 0 {
+				defaultValue = value
+			}
+		case reflect.Float64:
+			inputType = input.TypeDouble
+			inputKind = input.KindInput
+			if !isSecret && value.(float64) != 0.0 {
+				defaultValue = value
+			}
+		case reflect.Bool:
+			inputType = input.TypeBool
+			inputKind = input.KindConfirm
+			defaultValue = value
+		case reflect.Slice:
+			inputType = input.TypeStringArray
+			inputKind = input.KindMultiSelect
+			// Each element must be string
+			for i := 0; i < valRef.Len(); i++ {
+				item := valRef.Index(i)
+				// Unwrap interface
+				if item.Type().Kind() == reflect.Interface {
+					item = item.Elem()
+				}
+				// Check item type
+				if itemKind := item.Kind(); itemKind != reflect.String {
+					// Value is not array of strings
+					return
+				}
+				inputOptions = append(inputOptions, input.Option{
+					Id:   item.String(),
+					Name: item.String(),
+				})
+			}
+			if !isSecret && valRef.Len() > 0 {
+				defaultValue = value
+			}
+		default:
+			return
+		}
+
+		// Example
+		example := ""
+		if !isSecret {
+			example = strhelper.Truncate(cast.ToString(value), 20, "…")
+		}
+
+		// Add
+		d.addInputForField(objectKey, fieldPath, example, input.Input{
+			Id:      utils.NormalizeName(componentId.WithoutVendor() + "-" + fieldPath[1:].String()),
```

Comment:
`Id` vstupu ma default hodnotu podla komponenty a cesty ku hodnote,
napr. `ex-azure-cost-management-export-timeDimension-timeFrame`.
Autor sablony si to moze upravit podla seba.